airflow/providers/amazon/aws/sensors/s3.py

```diff
@@ -77,31 +77,24 @@ def __init__(
         **kwargs,
     ):
         super().__init__(**kwargs)
-
         self.bucket_name = bucket_name
         self.bucket_key = bucket_key
         self.wildcard_match = wildcard_match
         self.aws_conn_id = aws_conn_id
         self.verify = verify
         self.hook: Optional[S3Hook] = None
 
-    def poke(self, context: 'Context'):
-
+    def _resolve_bucket_and_key(self):
+        """If key is URI, parse bucket"""
         if self.bucket_name is None:
-            parsed_url = urlparse(self.bucket_key)
-            if parsed_url.netloc == '':
-                raise AirflowException('If key is a relative path from root, please provide a bucket_name')
-            self.bucket_name = parsed_url.netloc
-            self.bucket_key = parsed_url.path.lstrip('/')
+            self.bucket_name, self.bucket_key = S3Hook.parse_s3_url(self.bucket_key)
         else:
             parsed_url = urlparse(self.bucket_key)
```

Comment:
Doesn't have to happen in this PR but I wonder if `S3Hook.parse_s3_url()` could be augmented with some sort of configurable check between validating a URI vs. a relative path? Centralize some of this logic there without have `urlparse()` is different places. Anyway, just a thought.